appeals/web/src/server/lib/mappers/data/appeal/mapper.js

```diff
@@ -78,25 +82,35 @@ export async function initialiseAndMapAppealData(
 
 	const userHasUpdateCasePermission = userHasPermission(permissionNames.updateCase, session);
 
-	/** @type {Record<string, SubMapper>} */
-	const submappers = submaps[appealDetails.appealType];
-
-	/** @type {{appeal: MappedInstructions}} */
-	const mappedData = {
-		appeal: {}
+	return {
+		appealDetails,
+		currentRoute,
+		session,
+		skipAssignedUsersData: !!skipAssignedUsersData,
+		userHasUpdateCasePermission,
+		caseOfficerUser,
+		inspectorUser
 	};
+};
 
-	Object.entries(submappers).forEach(([key, submapper]) => {
-		mappedData.appeal[key] = submapper({
-			appealDetails,
-			currentRoute,
-			session,
-			skipAssignedUsersData,
-			userHasUpdateCasePermission,
-			caseOfficerUser,
-			inspectorUser
-		});
-	});
+/**
+ * @param {InnerParams} params
+ * @returns {string}
+ */
+const getAppealType = ({ appealDetails }) => {
+	if (appealDetails === undefined) {
+		throw new Error('appealDetails is undefined');
+	}
+
+	if (!appealDetails.appealType) {
+		throw new Error('No appealType on appealDetails');
+	}
+	return appealDetails.appealType;
+};
 
-	return mappedData;
-}
+export const initialiseAndMapAppealData = initialiseAndMapDataFactory(
+	getSubmapperParams,
```

Comment:
Would it be simpler to just pass the submapper params themselves rather than a function to generate them?

Reply:
That would also remove the need for `initialiseAndMapData` to be changed to async, I think

Reply:
I don't think this would simplify it. If we passes in params instead here's how I see that evolving: 
1. non-trivial param calculations for appeals mapper are copy and pasted into the 4 files where the mapper is used
2. this duplicate code looks messy, we create a function to create appeal submapper params
3. this function gets called in four files
The way I've done it here that function we'd end up creating is only needed in one place rather than in every place the mapper is used